src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -98,7 +98,22 @@ protected override ValueTask<RateLimitLease> WaitAsyncCore(int permitCount, Canc
                 // Don't queue if queue limit reached
                 if (_queueCount + permitCount > _options.QueueLimit)
```

Comment:
Is it possible _queueCount + permitCount could wrap to be < 0?

Reply:
Yep, fixed by using subtraction instead and added a test to cover that case.